tests/foreman/cli/test_host.py

```diff
@@ -660,6 +660,62 @@ def test_negative_register_twice(self):
             # host being already registered.
             self.assertEqual(result.return_code, 64)
 
+    def test_positive_register_twice_with_uppercase_chars_in_hostname(self):
+        """Register twice a client host that contain upper case chars in
+        hostname.
+
+        :id: 59c20379-b878-46ce-ad3e-ed6969ea6a5f
+
+        :steps:
+            1. Create a client host with upper case chars in hostname
+            2. register the host with command "subscription-manager register"
+            3. register the host a second time with command
+                "subscription-manager register --force"
+
+        :expectedresults: host registered the second time, without error
+
+        :BZ: 1361309
+
+        :CaseLevel: System
+        """
+        activation_key = make_activation_key({
+            'content-view-id': self.promoted_cv['id'],
+            'lifecycle-environment-id': self.new_lce['id'],
+            'organization-id': self.new_org['id'],
+        })
+        name = gen_string('alpha')
+        # make all the odd chars as upper case and the even one lower case
+        # to have a name like OpQrSTuVw
+        name_chars = list(name)
+        for i in range(len(name_chars)):
+            if i % 2 == 0:
+                name_chars[i] = name_chars[i].upper()
+            else:
+                name_chars[i] = name_chars[i].lower()
+        target_image = ''.join(name_chars)
```

Comment:
One-linear suggestion instead all these lines with conditions and double conversion:
```python
''.join((str.upper,str.lower)[i%2](ch) for i,ch in enumerate(name))
```
But maybe it's a bit too tricky, i'm not sure. Anyhow, not a blocker 🙂

Reply:
Or at least you could drop `else:`:
```python
name_chars = list(name.lower())
for i in range(len(name_chars)):
    if i % 2 == 0:
        name_chars[i] = name_chars[i].upper()
target_image = ''.join(name_chars)
```